awx/ui_next/src/screens/Project/shared/ProjectForm.jsx

```diff
@@ -0,0 +1,485 @@
+                type="text"
+                validate={required(null, i18n)}
+                tooltipMaxWidth="350px"
+                tooltip={scmUrlTooltips[values.scm_type]}
+              />
+            )}
+            {(values.scm_type === 'git' ||
+              values.scm_type === 'hg' ||
+              values.scm_type === 'svn') && (
+              <FormField
+                id="project-scm-branch"
+                name="scm_branch"
+                type="text"
+                label={scmBranchLabels[values.scm_type]}
+                tooltip={i18n._(t`Branch to checkout. In addition to branches,
+                    you can input tags, commit hashes, and arbitrary refs. Some
+                    commit hashes and refs may not be availble unless you also
+                    provide a custom refspec.`)}
+              />
+            )}
+            {values.scm_type === 'git' && (
+              <FormField
+                id="project-scm-refspec"
+                label={i18n._(t`SCM Refspec`)}
+                name="scm_refspec"
+                type="text"
+                tooltipMaxWidth="400px"
+                tooltip={
+                  <span>
+                    {i18n._(t`A refspec to fetch (passed to the Ansible git
+                        module). This parameter allows access to references via
+                        the branch field not otherwise available.`)}
+                    <br />
+                    <br />
+                    {i18n._(
+                      t`Note: This field assumes the remote name is "origin".`
+                    )}
+                    <br />
+                    <br />
+                    {i18n._(t`Examples include:`)}
+                    <ul style={{ margin: '10px 0 10px 20px' }}>
+                      <li>refs/*:refs/remotes/origin/*</li>
+                      <li>
+                        refs/pull/62/head:refs/remotes/origin/pull/62/head
+                      </li>
+                    </ul>
+                    {i18n._(t`The first fetches all references. The second
+                        fetches the Github pull request number 62, in this example
+                        the branch needs to be "pull/62/head".`)}
+                    <br />
+                    <br />
+                    {i18n._(t`For more information, refer to the`)}{' '}
+                    <Link to="https://docs.ansible.com/ansible-tower/latest/html/userguide/projects.html#manage-playbooks-using-source-control">
+                      {i18n._(t`Ansible Tower Documentation.`)}
+                    </Link>
+                  </span>
+                }
+              />
+            )}
+            {(values.scm_type === 'git' ||
+              values.scm_type === 'hg' ||
+              values.scm_type === 'svn') && (
+              <Field
+                name="credential"
+                render={({ form }) => (
+                  <CredentialLookup
+                    credentialTypeId={2}
```

Comment:
Credential type ids aren't fixed and can change - they're technically generated at install time. It's not safe to hard-code them. 

We'll probably want to look up the id of the credential type(s) dynamically using their name/namespace as part of the initialization of this view.

Reply:
👍 I'll update the form to grab the credential_type id based off of their kind/name.